pkg/addons/default.go

```diff
@@ -43,33 +43,37 @@ func Default(o unstructured.Unstructured) {
 	gk := o.GroupVersionKind().GroupKind()
 
 	switch gk.String() {
+	case "CronJob.batch":
+		for _, c := range jsonpath.MustCompile("$.spec.jobTemplate.spec.template.spec").Get(o.Object) {
+			defaultPodSpec(c.(map[string]interface{}))
+		}
+
+		jsonpath.MustCompile("$.spec.concurrencyPolicy").DeleteIfMatch(o.Object, "Allow")
+		jsonpath.MustCompile("$.spec.failedJobsHistoryLimit").DeleteIfMatch(o.Object, int64(1))
+		jsonpath.MustCompile("$.spec.successfulJobsHistoryLimit").DeleteIfMatch(o.Object, int64(3))
+		jsonpath.MustCompile("$.spec.suspend").DeleteIfMatch(o.Object, false)
+
 	case "DaemonSet.apps":
 		jsonpath.MustCompile("$.spec.revisionHistoryLimit").DeleteIfMatch(o.Object, int64(10))
 
 		for _, c := range jsonpath.MustCompile("$.spec.template.spec").Get(o.Object) {
 			defaultPodSpec(c.(map[string]interface{}))
 		}
 
-		defaultUpdateStrategy := map[string]interface{}{
-			"rollingUpdate": map[string]interface{}{
-				"maxUnavailable": int64(1),
-			},
-			"type": "RollingUpdate",
-		}
-		jsonpath.MustCompile("$.spec.updateStrategy").DeleteIfMatch(o.Object, defaultUpdateStrategy)
+		jsonpath.MustCompile("$.spec.updateStrategy.rollingUpdate.maxUnavailable").DeleteIfMatch(o.Object, int64(1))
+		jsonpath.MustCompile("$.spec.updateStrategy.rollingUpdate").DeleteIfMatch(o.Object, map[string]interface{}{})
```

Comment:
Why are we deleting strategy? It might be (and in some cases are) component specific?

Reply:
It gets defaulted to RollingUpdate for all workload controllers in apps/v1 (deployments, daemonsets, statefulsets).